packages/jellyfish-api-core/__tests__/category/loan/setLoanToken.test.ts

```diff
@@ -0,0 +1,280 @@
+      token: 'Token7',
+      currency: 'USD'
+    }], 1])
+    await testing.generate(1)
+
+    const loanTokenId = await testing.rpc.loan.setLoanToken({
+      symbol: 'Token7',
+      name: 'x'.repeat(129), // 129 letters
+      priceFeedId
+    })
+    await testing.generate(1)
+
+    const data = await testing.container.call('listloantokens', [])
+    const index = Object.keys(data).indexOf(loanTokenId) + 1
+    expect(data[loanTokenId].token[index].name).toStrictEqual('x'.repeat(128)) // Only remain the first 128 letters.
+  })
+
+  it('should setLoanToken if mintable is false', async () => {
+    const priceFeedId = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'Token8',
+      currency: 'USD'
+    }], 1])
+    await testing.generate(1)
+
+    const loanTokenId = await testing.rpc.loan.setLoanToken({
+      symbol: 'Token8',
+      priceFeedId,
+      mintable: false
+    })
+    expect(typeof loanTokenId).toStrictEqual('string')
+    expect(loanTokenId.length).toStrictEqual(64)
+    await testing.generate(1)
+  })
+
+  it('should setLoanToken if interest number is greater than 0 and has less than 9 digits in the fractional part', async () => {
+    const priceFeedId = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'Token9',
+      currency: 'USD'
+    }], 1])
+    await testing.generate(1)
+
+    const loanTokenId = await testing.rpc.loan.setLoanToken({
+      symbol: 'Token9',
+      priceFeedId,
+      interest: new BigNumber(15.12345678) // 8 digits in the fractional part
+    })
+    expect(typeof loanTokenId).toStrictEqual('string')
+    expect(loanTokenId.length).toStrictEqual(64)
+    await testing.generate(1)
+  })
+
+  it('should not setLoanToken if interest number is greater than 0 and has more than 8 digits in the fractional part', async () => {
+    const priceFeedId = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'Token10',
+      currency: 'USD'
+    }], 1])
+    await testing.generate(1)
+
+    const promise = testing.rpc.loan.setLoanToken({
+      symbol: 'Token10',
+      priceFeedId,
+      interest: new BigNumber(15.123456789) // 9 digits in the fractional part
+    })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Invalid amount\', code: -3, method: setloantoken')
+  })
+
+  it('should not setLoanToken if interest number is less than 0', async () => {
+    const priceFeedId = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'Token11',
+      currency: 'USD'
+    }], 1])
+    await testing.generate(1)
+
+    const promise = testing.rpc.loan.setLoanToken({
+      symbol: 'Token11',
+      priceFeedId,
+      interest: new BigNumber(-15.12345678)
+    })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Amount out of range\', code: -3, method: setloantoken')
+  })
+
```

Comment:
```suggestion

  it('should not setLoanToken if interest number is greater than 1200000000', async () => {
    const priceFeedId = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
      token: 'Token11',
      currency: 'USD'
    }], 1])
    await testing.generate(1)

    const promise = testing.rpc.loan.setLoanToken({
      symbol: 'Token11',
      priceFeedId,
      interest: new BigNumber('1200000000').plus('0.00000001')
    })
    await expect(promise).rejects.toThrow('RpcApiError: \'Amount out of range\', code: -3, method: setloantoken')
  })
 
```
Add this test for exhaustivity

Reply:
Good catch. I have added the test unit. The only thing I changed is the token / symbol. I used `Token12` instead.